GraphsV2/GaugeChart/index.js

```diff
@@ -0,0 +1,146 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { compose } from 'redux';
+import { Cell, PieChart, Pie } from 'recharts';
+
+import WithConfigHOC from '../../HOC/WithConfigHOC';
+import WithValidationHOC from '../../HOC/WithValidationHOC';
+import config from './default.config';
+import colorConvert from 'color-convert';
+import { RADIAN } from '../../constants';
+
+const renderCustomizedLabel = ({ cx, cy, midAngle, outerRadius, label }) => {
+    const radius = outerRadius + 10;
+    const x = cx + radius * Math.cos(-midAngle * RADIAN);
+    const y = cy + radius * Math.sin(-midAngle * RADIAN);
+
+    return (
+        <text x={x} y={y} fill="black" textAnchor="middle" dominantBaseline="central">
+            {label}
+        </text>
+    );
+};
+
+const Arrow = ({ cx, cy, midAngle, outerRadius, width, fontSize, chartValue }) => {
+    const sin = Math.sin(-RADIAN * midAngle);
+    const cos = Math.cos(-RADIAN * midAngle);
+    const mx = cx + (outerRadius + width * 0.03) * cos;
+    const my = cy + (outerRadius + width * 0.03) * sin;
+    return (
+        <g>
+            <text x={cx + 15} y={cy + 20} style={{ fontSize: fontSize }} fill="black" textAnchor={mx > cx ? 'start' : 'end'} dominantBaseline="central">
+                {parseInt(chartValue)}%
+          </text>
+            <circle cx={cx} cy={cy} r={width * 0.05} fill="#666" stroke="none" />
+            <path d={`M${cx},${cy}L${mx},${my}`} strokeWidth="6" stroke="#666" fill="none" strokeLinecap="round" />
+        </g>
+    );
+};
+
+let chartData = [];
```

Comment:
why do we add these variables here? If any of the data is changes then re-render will not happen

Reply:
Actually these variables are not intended to re-render the graph. The graph only re-renders when the props data changes.